.github/workflows/ephemeral-env.yml

```diff
@@ -1,145 +1,121 @@
+    name: Evaluate ephemeral env label trigger
     runs-on: ubuntu-24.04
     permissions:
       pull-requests: write
     outputs:
-      slash-command: ${{ steps.eval-body.outputs.result }}
+      slash-command: ${{ steps.eval-label.outputs.result }}
       feature-flags: ${{ steps.eval-feature-flags.outputs.result }}
+      sha: ${{ steps.get-sha.outputs.sha }}
     env:
       DOCKERHUB_USER: ${{ secrets.DOCKERHUB_USER }}
       DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
 
     steps:
-    - name: Debug
-      run: |
-        echo "Comment on PR #${{ github.event.issue.number }} by ${{ github.event.issue.user.login }}, ${{ github.event.comment.author_association }}"
+      - name: Check for the "testenv-up" label
+        id: eval-label
+        run: |
+          if [[ "${{ github.event.label.name }}" == "testenv-up" ]]; then
+            echo "result=up" >> $GITHUB_OUTPUT
+          else
+            echo "result=noop" >> $GITHUB_OUTPUT
+            exit 1
+          fi
 
-    - name: Eval comment body for /testenv slash command
-      uses: actions/github-script@v7
-      env:
-        COMMENT_BODY: ${{ github.event.inputs.comment_body || github.event.comment.body }}
-      id: eval-body
-      with:
-        result-encoding: string
-        script: |
-          const pattern = /^\/testenv (up|down)/;
-          const result = pattern.exec(process.env.COMMENT_BODY || '');
-          return result === null ? 'noop' : result[1];
+      - name: Get PR Info
+        uses: actions/github-script@v7
+        id: get-pr-info
+        with:
+          script: |
+            const pr = await github.rest.pulls.get({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: context.payload.pull_request.number,
+            });
+            return pr.data;
 
-    - name: Looking for feature flags
-      uses: actions/github-script@v7
-      env:
-        COMMENT_BODY: ${{ github.event.inputs.comment_body || github.event.comment.body }}
-      id: eval-feature-flags
-      with:
-        script: |
-          const pattern = /FEATURE_(\w+)=(\w+)/g;
-          let results = [];
-          [...process.env.COMMENT_BODY.matchAll(pattern)].forEach(match => {
-            const config = {
-              name: `SUPERSET_FEATURE_${match[1]}`,
-              value: match[2],
-            };
-            results.push(config);
-          });
-          return results;
+      - name: Check if a commit was pushed after PR was labeled
+        id: get-sha
+        env:
+          COMMENT_AT: ${{ github.event.pull_request.updated_at }}
+          PUSHED_AT: ${{ fromJSON(steps.get-pr-info.outputs.result).pushed_at }}
+        run: |
+          if [[ $(date -d "$PUSHED_AT" +%s) -gt $(date -d "$COMMENT_AT" +%s) ]]; then
+           echo "Commit was pushed after the PR was updated."
+           exit 1
+          fi
+          echo "sha=${{ fromJSON(steps.get-pr-info.outputs.result).head.sha }}" >> $GITHUB_OUTPUT
```

Comment:
If you're verifying the label, there is no need for this extra step. Just use github.event.pull_request.head.sha

Reply:
unsure if `github.event.pull_request.updated_at` could be greater then the time of the last push

Reply:
No, it’s the SHA at the time of label, which is the most secure

Reply:
`GITHUB_RUN_ID` can we useful at times too, not sure if it's better than SHA here, but would allow concurrent runs on the same SHA (do we want that? how do we want to handle potential SHA collisions?)

Reply:
I see we have the `concurrency` GHA thing set up properly, meaning a new request will kill the previous ones, which I believe is what we want. Using the `GITHUB_RUN_ID` as a unique id on the AWS-side could help segment/isolate/clarify the runs.

Reply:
I guess we need something deterministic that we can use to kill previous instances on PR close or unlabelling on the AWS side, so `GITHUB_RUN_ID` isn't the best for that ...

Reply:
Side note, I created `supersetbot` to make it easier to develop/test/deploy things outside of GHA. We're pretty deep already here, but we could move the bulk of the logic to `supersetbot ephemeral deploy {image_id}`, `supersetbot ephemeral stop {image_id}` ...

Reply:
concurrency seems fine to me (pushed a fix). it's unique at the PR level, so label events for new commits will cancel previous runs.

+1 on using supersetbot ephemeral deploy